mlir/tools/mlir-tblgen/OpDefinitionsGen.cpp

```diff
@@ -3583,6 +3583,22 @@ void OpEmitter::genTypeInterfaceMethods() {
   fctx.addSubst("_ctxt", "context");
   body << "  ::mlir::Builder odsBuilder(context);\n";
 
+  // Preprocessing stage to verify all accesses to operands are valid.
+  int maxAccessedIndex = -1;
+  for (int i = 0, e = op.getNumResults(); i != e; ++i) {
+    const InferredResultType &infer = op.getInferredResultType(i);
+    if (!infer.isArg())
+      continue;
+    auto arg = op.getArgToOperandOrAttribute(infer.getIndex());
```

Comment:
spell `auto` here please

Reply:
Apparently, even `getArgToOperandOrAttribute ` is declared with `auto`, so no one wants to expose return type.

Reply:
I see it declared as `  OperandOrAttribute getArgToOperandOrAttribute(int index) const;` ?

Reply:
my bad. I thought it's pure automatic type deduction, so didn't look for `->`